Adding explanations for fairness metrics in assessment.rst (#1111)
fairlearn/fairlearn: docs/user_guide/assessment.rst

    >>> print("ratio in recall = ", grouped_metric.ratio(method='between_groups'))    
    ratio in recall =  0.0

> Fairness metrics

Author: I liked "common fairness metrics" so that's what I've replaced the header with, but others chime in too!

----------------

> Parity constraints

Author: Removed the _Parity constraints_ subheader.

^^^^^^^^^^^^^^^^^^
In the sections below, we review the most common fairness metrics, as well
as their underlying assumptions and suggestions for use. Each metric requires
that some aspects of the predictor behavior be comparable across groups. In 
the mathematical definitions below, let :math:`X` denote a feature vector 
used for predictions, :math:`A` be a single sensitive feature (such as age 
or race), and :math:`Y` be the true label. Parity metrics are phrased in 
terms of expectations with respect to the distribution over :math:`(X,A,Y)`.

Demographic parity
~~~~~~~~~~~~~~~~~~

> Demographic parity is also known as *statistical parity*. A classifier

Author: Added the note!

> :math:`h` satisfies demographic parity under a distribution over

Author: Done! 

> Demographic parity is a fairness metric whose goal is to ensure a machine learning model's predictions are independent of membership in a sensitive group. In other words, demographic parity is achieved when the probability of a certain prediction is not dependent on sensitive group membership. In the binary classification scenario, demographic parity refers to equal selection rates across groups. For example, in the context of a resume screening model, equal selection would mean that the proportion of applicants selected for a job interview should be equal across groups.

[Formulas]

> Failing to achieve demographic parity could generate allocation harms. Allocation harms occur when AI systems allocate 
opportunities, resources, or information differently across different groups (for example, an AI hiring system that is more likely to advance resumes of male applicants than resumes of female applicants regardless of qualification). 
Demographic parity can be used to assess the extent of allocation harms because it reflects an assumption that resources should be allocated proportionally across groups. Of the metrics described in this section, it can be the easiest 
to implement. However, operationalizing fairness using demographic parity rests on a few assumptions: that either the dataset is not a good representation  of what the world actually looks like (e.g., a resume assessment system that is 
more likely to filter out qualified female applicants due to an organizational bias towards male applicants, regardless of skill level), or that the dataset is an accurate representation of the world, but the world is unjust. In reality, this may not be the case. The dataset might be an accurate representation of what the world actually looks like, or the process being modeled may not be unjust.

I'll think through additional examples for the underlying assumptions. They were mentioned in the issue, but I'd appreciate it if you could expand on what you were thinking here a little more. Thank you!

:math:`(X, A, Y)` if its prediction :math:`h(X)` is statistically
independent of the sensitive feature :math:`A`. This is equivalent to
:math:`\E[h(X) \given A=a] = \E[h(X)] \quad \forall a`. :footcite:`agarwal2018reductions`

> In the case of regression, a predictor :math:`f` satisfies demographic parity

Reviewer: Note that in the Fairlearn API, `demographic_parity_difference` is only defined for the classification scenario. I do think it is worthwhile to mention regression, but it might be better off as a separate paragraph within the `demographic parity` section, where we highlight that you can compute `mean_prediction()` across groups as one way to measure allocation harms in the regression context. Again, adding examples is important for users to understand why this might matter.

Author: I added another paragraph underneath the classification scenario.

under a distribution over :math:`(X, A, Y)` if :math:`f(X)` is independent
of the sensitive feature :math:`A`. This is equivalent to
:math:`\P[f(X) \geq z \given A=a] = \P[f(X) \geq z] \quad \forall a, z`.
:footcite:`agarwal2019fair`

Demographic parity can be used to assess the extent of allocation harms, as it 
reflects an assumption that resources should be allocated proportionally 
across groups. Of the metrics described in this section, it can be the easiest 
to implement. However, operationalizing fairness using demographic parity 

> rests on assumptions: that either the dataset is not a good representation of 

Reviewer: I think these assumptions could be elaborated on further.

what the world actually looks like (e.g., a resume assessment system that is 
more likely to filter out qualified female applicants due to an organizational 
bias towards male applicants, regardless of skill level), or that the dataset 
is an accurate representation of the world, but the world is unjust. 

It's worth considering whether the assumptions underlying this measurement 

> model maintain construct validity (see :ref:`construct_validity`). 

Reviewer: In my view the more obvious connection to construct validity might be the construct validity of the target variable (i.e., the first assumption). The construct validity of demographic parity as a metric holds more generally for all fairness metrics, so perhaps we should add it to a section that's more generally about limitations of group fairness metrics? 

On a separate note, I think this paragraph might be a bit confusing for readers who are unfamiliar with issues around construct validity (which I suspect will be most readers). We should definitely refer readers to our user guid on the topic, but perhaps we can think of a way to explain this point in a more data scientist-friendly way, and then say something like "In the social sciences, this is referred to as construct validity, you can learn more about it here"

Author: I think the inclusion of a construct validity section specific to all of the fairness metrics might be helpful. I did add a few sentences introducing the construct validity topic to help clear things up.

Consequential validity, for example, asks how the world shaped by using the 
measurement model; in this case, the relevant question would be whether 
demographic parity meets the criteria for establishing "fairness", itself 
an unobservable theoretical construct, and whether satisfying demographic 
parity brings us closer to the world we'd like to see. 

> 

Author: I added the paragraph below and will add the Fair ML book as a citation if it's not in refs.bib already. Please let me know your thoughts! 

> Fairness metrics like demographic parity can also be used as optimization constraints during the machine learning model training process. However, demographic parity is not well-suited for this purpose because it does not place requirements on the exact distribution of predictions with respect to other important variables. To understand this concept further, consider an example from the Fairness in Machine Learning textbook :footcite:`fairml_book`. Suppose a company wishes to hire applicants from group A and B at the same rate. It carefully selects applicants from group A, but carelessly selects applicants from group B. In this scenario, even though members from both groups are selected at the same rate, it's likely that underqualified or unqualified applicants are chosen more frequently for one group (group B) than another. If applicants from group B consequently perform worse in comparison to applicants from group A, negative perceptions of their performance may form. If group B experiences higher levels of prejudice or discrimination, then these perceptions may reinforce existing negative stereotypes.

Equalized odds
~~~~~~~~~~~~~~

> A classifier :math:`h` satisfies equalized odds under a

Reviewer: Similar as with previous paragraph, I'd suggest to start with the more intuitive explanation, i.e., we want the model to perform equally well for different groups, then why we would want that (e.g., to avoid allocation / quality of service harms).

Author: Done! Please let me know your thoughts.

> The goal of the equalized odds fairness metric is to ensure a machine learning model performs equally well for different groups. It is stricter than demographic parity because it requires that the machine learning model's predictions are not only independent of sensitive group membership, but that groups have the same false positive rates and and true positive 
rates. This distinction is important because a model could achieve demographic parity (i.e., its predictions could be independent of sensitive group membership), but still generate more false positive predictions for one group versus others. Equalized odds does not create the selection issue discussed in the demographic parity section above. For example, in the hiring scenario where the goal is to choose applicants from group A and group B, ensuring the model performs equally well at choosing applicants from group A and B can circumvent the issue of the model optimizing by selecting applicants from one group at random.

distribution over :math:`(X, A, Y)` if its prediction :math:`h(X)` is
conditionally independent of the sensitive feature :math:`A` given the label
:math:`Y`. This is equivalent to
:math:`\E[h(X) \given A=a, Y=y] = \E[h(X) \given Y=y] \quad \forall a, y`.
:footcite:`agarwal2018reductions` Equalized odds requires that the true 
positive rate, :math:`\P(h(X)=1 | Y=1`, and the false positive rate, 
:math:`\P(h(X)=1 | Y=0`, be equal across groups. 

The inclusion of false positive rates acknowledges that different groups 
experience different costs from misclassification. For example, in the case of 
a model predicting a negative outcome (e.g., probability of recidivating) 
that already disproportionately affects members of minority communities, 
false positive predictions reflect preexisting disparities in outcomes 
across minority and majority groups. Equalized odds further enforces that the 
accuracy is equally high across all groups, punishing models that only 
perform well on majority groups.

Equalized odds and can be used to diagnose both allocation harms as well as 

> quality-of-service harms. It can be useful for diagnosing allocation harms 

Reviewer: I would given an example of allocation harm (e.g., the typical example of hiring) and quality of service harm (e.g., gender shades). For allocation harms, I think it is good to highlight how this metric differs from demographic parity (i.e., equalized odds considers not the allocation of resources, but, for lack of a better term, the allocation of "deserved" resources, where "deserved" is determined by the target variable). 

I think this will make the point of the assumption that the target variable is a good measurement of the phenomena being modeled clearer as well.

Author: Added the following details with new text in bold:

> Equalized odds and can be used to diagnose both allocation harms as well as quality-of-service harms. **Allocation harms occur when AI systems allocate opportunities, resources, or information differently across different groups (for example, an AI hiring system that is more likely to advance resumes of male applicants than resumes of female applicants regardless of qualification). Quality-of-service harms occur when an AI system does not work as well for one group versus another (for example, facial recognition systems that are more likely to fail for dark-skinned individuals). For more information on AI harms, see 
:ref:`types_of_harms`.** 

Equalized odds can be useful for diagnosing allocation harms 
because its goal is to ensure that a machine learning model works equally well for different groups. **Another way to think about equalized odds is to contrast it with demographic parity. While demographic parity assesses the allocation of resources generally, equalized odds focuses on the allocation of resources that were actually distributed to members of that group (indicated by the positive target variable Y=1).** However, equalized odds makes the assumption that the target variable :math:`Y` is a good measurement of the phenomena being modeled, but that assumption may not hold if the measurement does not satisfy the requirements of construct validity.

because its goal is to ensure that a machine learning model works equally 
well for different groups. However, equalized odds makes the assumption 
that the target variable :math:`Y` is a good measurement of the phenomena 
being modeled, but that assumption may not hold if the measurement does not 
satisfy the requirements of construct validity.

Equal opportunity
~~~~~~~~~~~~~~~~~
Equal opportunity is a relaxed version of equalized odds that only considers
conditional expectations with respect to positive labels, i.e., :math:`Y=1`.
:footcite:`hardt2016equality` Another way of thinking about this metric is 
requiring equal outcomes only within the subset of records belonging to the 
positive class. For example, in the hiring example, equal opportunity 
requires that the individuals who are actually hired have an equal opportunity 
of being hired in the first place. However, by not considering whether false 
positive rates are equivalent across groups, equal opportunity does not 
capture the costs of missclassification disparities.

> Considerations for conditional group fairness

Reviewer: Nitpick: this paragraph might make even more sense as a subparagraph of demographic parity. Though I suppose you could make similar arguments for equalized odds, it seems most natural to consider for demographic parity.

Author: Yeah, should we move it up to the demographic parity section?

Reviewer: Sounds good!

Author: Ok, I will move the conditional group fairness section up to be a subsection of demographic parity.

admissions to the University of California, Berkley :footcite:`bickel1975biasinadmissions`. 
The study showed that when observing admissions by gender, men applying were 
more likely than women to be accepted. However, drilling down into admissions 
by department revealed that women tended to apply to departments with more 
competitive admissions requirements, whereas men tended to apply to less 
competitive depoartments. The more granular analysis showed only four out of 
85 departments exhibited bias against women, and six departments exhibited 
bias towards men. In general, the data indicated departments exhibited a bias 
in favor of minority-gendered applicants, which is opposite from the trend 
observed in the aggregate data.

This phenomenon is important to fairness evaluation because metrics like 
demographic parity may be different when calculated at an aggregate level and  
within more granular categories. In the case of demographic parity, we might 
need to review 
:math:`\E[h(X) \given A=a, D=d] = \E[h(X) \given D=d] \quad \forall a` 
where D represents the feature(s) within X across which members of the groups 

> within A are distributed. :footcite:`xu2020conditionalfairness` 

Author: Ah ok, I've removed the reference for now.

Demographic parity would then require that the prediction of the target  
variable is statistically independent of sensitive attributes conditional 
on D. Simply aggregating outcomes across high-level categories can be 
misleading when the data can be further disaggregated. It's important to 
review metrics across these more graular categories, if they exist, 

> to verify that disparate outcomes persist across all levels of aggregation.

Reviewer: We should probably mention the caveat that more granular categories also means smaller sample sizes and the problem of multiple comparisons.

Author: To address the first point, I added the following sentence. But I think there's a clearer way to phrase this. 
> More granular categories generally contain smaller sample sizes, and 
it can be more difficult to establish that trends seen in very small 
samples are not due to random chance.  

How much detail should we get into on multiple comparisons? It seems like there are statistical methods meant to deal with this, but I'm unsure if this is out of scope for the assessment section.

of a certain prediction is not dependent on sensitive group membership. In 
the binary classification scenario, demographic parity refers to equal 
selection rates across groups. For example, in the context of a resume 
screening model, equal selection would mean that the proportion of 
applicants selected for a job interview should be equal across groups.

We mathematically define demographic parity using the following 
set of equations.
A classifier :math:`h` satisfies demographic parity under a distribution 
over :math:`(X, A, Y)` if its prediction :math:`h(X)` is statistically
independent of the sensitive feature :math:`A`. This is equivalent to
:math:`\E[h(X) \given A=a] = \E[h(X)] \quad \forall a`. :footcite:`agarwal2018reductions`

In the case of regression, a predictor :math:`f` satisfies demographic parity
under a distribution over :math:`(X, A, Y)` if :math:`f(X)` is independent
of the sensitive feature :math:`A`. This is equivalent to
:math:`\P[f(X) \geq z \given A=a] = \P[f(X) \geq z] \quad \forall a, z`.
:footcite:`agarwal2019fair` Another way to think of demographic parity in a 
regression scenario is to compare the average predicted value across groups.

> Note that in the Faiurlearn API, :func:`fairlearn.metrics.demographic_parity_difference` 

Reviewer: ```suggestion
Note that in the Fairlearn API, :func:`fairlearn.metrics.demographic_parity_difference` 
```

> Note that in the Faiurlearn API, :method:`fairlearn.metrics.demographic_parity_difference` 

Reviewer: ```suggestion
Note that in the Faiurlearn API, :func:`fairlearn.metrics.demographic_parity_difference` 
```

   Demographic parity is also sometimes referred to as *independence*, *group fairness*, *statistical parity*, and *disparate impact*.

Failing to achieve demographic parity could generate allocation harms. 
Allocation harms occur when AI systems allocate 
opportunities, resources, or information differently across different 
groups (for example, an AI hiring system that is more likely to advance resumes 
of male applicants than resumes of female applicants regardless of qualification). 
Demographic parity can be used to assess the extent of allocation harms because it 
reflects an assumption that resources should be allocated proportionally 
across groups. Of the metrics described in this section, it can be the easiest 
to implement. However, operationalizing fairness using demographic parity 
rests on a few assumptions: that either the dataset is not a good representation  
of what the world actually looks like (e.g., a resume assessment system that is 
more likely to filter out qualified female applicants due to an organizational 
bias towards male applicants, regardless of skill level), or that the dataset 
is an accurate representation of the phenomena being modeled, but the 

> phenomena itself is unjust. In reality, this may not be the case. The

Reviewer 2: Can we add a parenthetical example to help readers understand what it might mean for "the phenomena itself [to be] unjust"? Maybe something about "predictive policing"? There may be language in the user guide sections on construct validity (specifically consequential validity) or abstraction traps (formalism or framing traps, specifically) that might be helpful here (or you may want to give a quick parenthetical example and then link to one of those sections)

Author: I added the following example in bold: 

> However, operationalizing fairness using demographic parity rests on a few assumptions: that either the dataset is not a good representation  of what the world actually looks like (e.g., a resume assessment system that is more likely to filter out qualified female applicants due to an organizational bias towards male applicants, regardless of skill level), or that the dataset is an accurate representation of the phenomena being modeled, but the phenomena itself is unjust **(e.g., consider the case of predictive policing, where a system created to predict crime rates may correctly predict higher crime rates for certain areas, but simultaneously fail to consider that those higher rates may be caused by disproportionate policing and overcrimininalization of those areas).** 

Is this what you're thinking of?

Reviewer 2: Looks great! thanks!!

dataset might be an accurate representation of the phenomena itself, 
or the phenomena being modeled may not be unjust. 
If either assumption is not true, then demographic parity may not provide 
a meaningful or useful measurement of the fairness of a model's predictions. 

Fairness metrics like demographic parity can also be used as optimization 
constraints during the machine learning model training process. However, 
demographic parity may not be well-suited for this purpose because 
it does not place requirements on the exact distribution of predictions with 
respect to other important variables. To understand this concept further, 
consider an example from the Fairness in Machine Learning textbook :footcite:`fairml_book`. 

    "However, decisions based on a classifier that satisfies independence can 
    have undesirable properties (and similar arguments apply to other 
    statistical critiera). Here is one way in which this can happen, 
    which is easiest to illustrate if we imagine a callous or ill-intentioned 

>     decision maker. Imagine a company that in group a hires diligently 

Reviewer 2: Could we italicize or capitalize the "group a" and "group b" so it's a little bit easier to read?

Author: Done!

    selected applicants at some rate p>0. In group bbb, the company 
    hires carelessly selected applicants at the same rate p. Even though 
    the acceptance rates in both groups are identical, it is far more likely 
    that unqualified applicants are selected in one group than in the other. 
    As a result, it will appear in hindsight that members of group b 
    performed worse than members of group a, thus establishing a negative 
    track record for group b."

It's also worth considering whether the assumptions underlying demographic
parity maintain construct validity (see :ref:`construct_validity`). 

> Construct validity is a concept in the social sciences that assess the 

Reviewer 2: ```suggestion
Construct validity is a concept in the social sciences that assesses the 
```

extent to which the ways we choose to measure abstract 

> phenomena is valid. For demographic parity, one relevant question would be 

Reviewer 2: ```suggestion
phenomena are valid. For demographic parity, one relevant question would be 
```

whether demographic parity meets the criteria for establishing "fairness", 
itself an unobservable theoretical construct. Further, it's important 
to ask whether satisfying demographic parity actually brings us closer 
to the world we'd like to see. 

.. _conditional_group_fairness:

> In some cases, we may observe a trend in several groups of data, but that 

Reviewer 2: Could we re-word "several groups of data" to "data from multiple demographic groups"? It may make the rest of this a bit clearer (unless that isn't what you mean here)

Author: Implemented!

by department revealed that women tended to apply to departments with more 
competitive admissions requirements, whereas men tended to apply to less 
competitive depoartments. The more granular analysis showed only four out of 
85 departments exhibited bias against women, and six departments exhibited 
bias towards men. In general, the data indicated departments exhibited a bias 
in favor of minority-gendered applicants, which is opposite from the trend 
observed in the aggregate data.

This phenomenon is important to fairness evaluation because metrics like 
demographic parity may be different when calculated at an aggregate level and  
within more granular categories. In the case of demographic parity, we might 
need to review 
:math:`\E[h(X) \given A=a, D=d] = \E[h(X) \given D=d] \quad \forall a` 
where D represents the feature(s) within X across which members of the groups 
within A are distributed. 
Demographic parity would then require that the prediction of the target  
variable is statistically independent of sensitive attributes conditional 
on D. Simply aggregating outcomes across high-level categories can be 
misleading when the data can be further disaggregated. 

> More granular categories generally contain smaller sample sizes, and 

Reviewer 2: Could we move this sentence ("More granular categories...") down to the start of the next paragraph (maybe starting it with a "However" to indicate it's about issues)? So that whole next paragraph would then be about statistical limitations or potential threats to validity - e.g., small sample sizes and multiple comparisons.

Does that work for you?

Author: Done!

For example, in the hiring scenario where the goal is to choose applicants 
from group A and group B, ensuring the model performs equally well at 
choosing applicants from group A and B can circumvent the issue of the model 
optimizing by selecting applicants from one group at random.

We mathematically define equalized odds using the following 
set of equations. A classifier :math:`h` satisfies equalized 
odds under a distribution over :math:`(X, A, Y)` if its 
prediction :math:`h(X)` is
conditionally independent of the sensitive feature :math:`A` given the label
:math:`Y`. This is equivalent to
:math:`\E[h(X) \given A=a, Y=y] = \E[h(X) \given Y=y] \quad \forall a, y`.
:footcite:`agarwal2018reductions` Equalized odds requires that the true 
positive rate, :math:`\P(h(X)=1 | Y=1`, and the false positive rate, 
:math:`\P(h(X)=1 | Y=0`, be equal across groups. 

The inclusion of false positive rates acknowledges that different groups 
experience different costs from misclassification. For example, in the case of 
a model predicting a negative outcome (e.g., probability of recidivating) 
that already disproportionately affects members of minority communities, 

> false positive predictions reflect preexisting disparities in outcomes 

Reviewer 2: ```suggestion
false positive predictions reflect pre-existing disparities in outcomes 
```

across minority and majority groups. Equalized odds further enforces that the 
accuracy is equally high across all groups, punishing models that only 
perform well on majority groups.

If a machine learning model does not perform equally well for all groups, 
then it could generate allocation or quality-of-service harms.

> Equalized odds and can be used to diagnose both allocation harms as well as 

Reviewer 2: ```suggestion
Equalized odds can be used to diagnose both allocation harms as well as 
```

    >>> print("ratio in recall = ", grouped_metric.ratio(method='between_groups'))    
    ratio in recall =  0.0

Common fairness metrics
-----------------------
In the sections below, we review the most common fairness metrics, as well
as their underlying assumptions and suggestions for use. Each metric requires
that some aspects of the predictor behavior be comparable across groups. In 

> the mathematical definitions below, let :math:`X` denote a feature vector 

Author: Done!

used for predictions, :math:`A` be a single sensitive feature (such as age 

> or race), and :math:`Y` be the true label. Parity metrics are phrased in 

Author: Edited to "fairness" metrics, which is in line with how the paragraph starts.

group. In other words, demographic parity is achieved when the probability 
of a certain prediction is not dependent on sensitive group membership. In 
the binary classification scenario, demographic parity refers to equal 
selection rates across groups. For example, in the context of a resume 
screening model, equal selection would mean that the proportion of 
applicants selected for a job interview should be equal across groups.

We mathematically define demographic parity using the following 
set of equations.
A classifier :math:`h` satisfies demographic parity under a distribution 
over :math:`(X, A, Y)` if its prediction :math:`h(X)` is statistically
independent of the sensitive feature :math:`A`. This is equivalent to
:math:`\E[h(X) \given A=a] = \E[h(X)] \quad \forall a`. :footcite:`agarwal2018reductions`

In the case of regression, a predictor :math:`f` satisfies demographic parity
under a distribution over :math:`(X, A, Y)` if :math:`f(X)` is independent
of the sensitive feature :math:`A`. This is equivalent to
:math:`\P[f(X) \geq z \given A=a] = \P[f(X) \geq z] \quad \forall a, z`.
:footcite:`agarwal2019fair` Another way to think of demographic parity in a 

> regression scenario is computing the `mean_prediction()` accross groups 

Reviewer: ```suggestion
regression scenario is to compare the average predicted value across groups.
```

and determining how different mean predictions are for different groups. 

> Note that in the Faiurlearn API, :method:`demographic_parity_difference` 

Reviewer: Could you check whether this will actually refer to the right function, or should we use something like :method:`fairlearn.metrics.demographic_parity_difference` ?

Author: Ah, my apologies for the incorrect reference, it's been edited.

is only defined for classification. 

.. note::
   Demographic parity is also sometimes referred to as *independence*, *group fairness*, *statistical parity*, and *disparate impact*.

Failing to achieve demographic parity could generate allocation harms. 
Allocation harms occur when AI systems allocate 
opportunities, resources, or information differently across different 
groups (for example, an AI hiring system that is more likely to advance resumes 
of male applicants than resumes of female applicants regardless of qualification). 
Demographic parity can be used to assess the extent of allocation harms because it 
reflects an assumption that resources should be allocated proportionally 
across groups. Of the metrics described in this section, it can be the easiest 
to implement. However, operationalizing fairness using demographic parity 
rests on a few assumptions: that either the dataset is not a good representation  
of what the world actually looks like (e.g., a resume assessment system that is 
more likely to filter out qualified female applicants due to an organizational 
bias towards male applicants, regardless of skill level), or that the dataset 

> is an accurate representation of the world, but the world is unjust. 

Author: Done!

In reality, this may not be the case. The dataset might be an accurate representation 
of what the world actually looks like, or the phenomena being modeled may not be 
unjust. If either assumption is not true, then demographic parity may not provide 
a meaningful or useful measurement of the fairness of a model's predictions. 

Fairness metrics like demographic parity can also be used as optimization 
constraints during the machine learning model training process. However, 
demographic parity is not well-suited for this purpose because 
it does not place requirements on the exact distribution of predictions with 
respect to other important variables. To understand this concept further, 

> consider an example from the Fairness in Machine Learning textbook :footcite:`fairml_book`. 

Author: Done!

Suppose a company wishes to hire applicants from group A and B at the 
same rate. It carefully selects applicants from group A, but carelessly 
selects applicants from group B. 
In this scenario, even though members from both groups are selected at 
the same rate, it's likely that underqualified or unqualified applicants 
are chosen more frequently for one group (group B) than another. 
If applicants from group B consequently perform worse in 
comparison to applicants from group A, negative perceptions of their 
performance may form. If group B experiences higher levels of prejudice 
or discrimination, then these perceptions may reinforce existing 
negative stereotypes. 

> It's also worth considering whether the assumptions underlying demographic

Author: Ah gotcha. I saved the initial two sentences and removed everything before "For demographic parity, one relevant question would be...". So now the two paragraphs have been reduced down to one.

> It's also worth considering whether the assumptions underlying demographic

Reviewer 2: Works for me! thanks!

in favor of minority-gendered applicants, which is opposite from the trend 
observed in the aggregate data.

This phenomenon is important to fairness evaluation because metrics like 
demographic parity may be different when calculated at an aggregate level and  
within more granular categories. In the case of demographic parity, we might 
need to review 
:math:`\E[h(X) \given A=a, D=d] = \E[h(X) \given D=d] \quad \forall a` 
where D represents the feature(s) within X across which members of the groups 
within A are distributed. 
Demographic parity would then require that the prediction of the target  
variable is statistically independent of sensitive attributes conditional 
on D. Simply aggregating outcomes across high-level categories can be 
misleading when the data can be further disaggregated. 
More granular categories generally contain smaller sample sizes, and 
it can be more difficult to establish that trends seen in very small 
samples are not due to random chance. 
It's important to review metrics across these more graular categories, 
if they exist, to verify that disparate outcomes persist across all levels 
of aggregation. 

> We also recommend watching out for the multiple comparisons problem, 

Reviewer 2: Can we link to an explanation of this?

Author: Done! Linked to this site: https://stats.libretexts.org/Bookshelves/Applied_Statistics/Book%3A_Biological_Statistics_(McDonald)/06%3A_Multiple_Tests/6.01%3A_Multiple_Comparisons

which states that the more statistical inferences are made, the 
more erroneous those inferences will become. For example, in the case 
of evaluating fairness metrics on multiple groups, as we break the 
groups down into more granular categories and evaluate those smaller 
groups, it will become more likely that these subgroups will 
differ enough to fail one of the metrics. For dealing with the multiple 
comparisons problem, we recommend investigating statistical techniques 

> meant to correct the errors produced by individual statistical tests. 

Reviewer 2: Can we add a link to a suggested technique? e.g., Bonferroni, Benjamini-Hochberg, etc

Author: Yes, linked to this site: https://www.statology.org/bonferroni-correction/
